### PR TITLE
Suspicious Funding - True Positive list fetching

### PR DESCRIPTION
**Pull request by RCantu92**
Adding logic to allow the _Suspicious Funding_ bot to fetch a list of True Positive Attacker EOAs and generate alerts of `Critical` severity and `Exploit` type when they're involved.

### CODE REVIEW

File: suspicious-funding-ts/src/agent.ts [hunk 1]
```diff
@@ -87,6 +95,19 @@ export const provideHandleTransaction =
     return findings;
   };
 
+export const provideHandleBlock = (fetcher: TruePositiveFetcher) => async (blockEvent: BlockEvent) => {
+  const findings: Finding[] = [];
+
+  // Choosing the denomincator based on chain's block time
```
Comment (Vxatz) [inline suspicious-funding-ts/src/agent.ts:101]: Just fixing a typo

```suggestion
  // Choosing the denominator based on chain's block time
```

File: suspicious-funding-ts/src/truePositive.fetcher.spec.ts [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+        expect(spy).toHaveBeenCalledWith("Both True Positive List fetching functions failed. Error: getTruePositiveListLocally() failed.");
+        // Confirm it registers `console.log`s inside of the nested `try/catch`s.
+        expect(spy).toHaveBeenCalledTimes(2);
+        expect(mockAttackers.has(mockAttacker)).toBe(false);
+        expect(mockAttackers.size).toBe(0)
+    });
+
+    it("will handle errors on both True Positive List fetching functions by not fetching anything - `error()` fallback invoked", async () => {
+        // This result will cause the `getTruePositiveListRemotely` to error out,
+        // then proceed to `getTruePositiveListLocally`, , which will also error out.
```
Comment (Vxatz) [inline suspicious-funding-ts/src/truePositive.fetcher.spec.ts:201]:
```suggestion
        // then proceed to `getTruePositiveListLocally`, which will also error out.
```

File: suspicious-funding-ts/src/utils.ts [hunk 3]
```diff
@@ -57,9 +57,9 @@ export const createFinding = (
   return Finding.fromObject({
     name: "Suspicious Funding Alert",
```
Comment (Vxatz) [inline suspicious-funding-ts/src/utils.ts:58]: Perhaps use "Malicious" here as well?

Comment (RCantu92) [reply suspicious-funding-ts/src/utils.ts:58]: That crossed my mind. I updated it, but now I'm wondering whether we should rename the bot. I think it _could_, but anything else we should keep in mind in regards to doing so?

I know for downstream purposes, the bot is referred to by its `botId`. But just checking with you.

Comment (Vxatz) [reply suspicious-funding-ts/src/utils.ts]: Hmm I don't think we really need to, unless it's specifically brought up to us

File: suspicious-funding-ts/src/agent.ts [hunk 4]
```diff
@@ -87,6 +95,19 @@ export const provideHandleTransaction =
...
   };
 
+export const provideHandleBlock = (fetcher: TruePositiveFetcher) => async (blockEvent: BlockEvent) => {
+  const findings: Finding[] = [];
+
+  // Choosing the denominator based on chain's block time
+  // so as to update TP list approx. once daily
+  const DAILY_BLOCKS_DENOMINATOR = chainId == 1 ? ETH_BLOCKS_IN_ONE_DAY : THREE_SECOND_BLOCKS_IN_ONE_DAY;
+  if (blockEvent.blockNumber % DAILY_BLOCKS_DENOMINATOR == 0) {
+      fetcher.getTruePositiveList(attackers);
```
Comment (Vxatz) [inline suspicious-funding-ts/src/agent.ts:105]:
```suggestion
      await fetcher.getTruePositiveList(attackers);
```

**Review by Vxatz — APPROVED** [review]
Only added an `await`, other than that it LGTM 👍

**Pull request merged**